components/style/values/specified/list.rs

```diff
@@ -9,8 +9,70 @@ use parser::{Parse, ParserContext};
 use std::fmt;
 use style_traits::{ParseError, StyleParseErrorKind, ToCss};
 use values::{Either, None_};
+#[cfg(feature = "gecko")]
+use values::CustomIdent;
+#[cfg(feature = "gecko")]
+use values::generics::CounterStyleOrNone;
 use values::specified::UrlOrNone;
 
+/// Specified and computed `list-style-type` property.
+#[cfg(feature = "gecko")]
+#[derive(Clone, Debug, Eq, MallocSizeOf, PartialEq, ToComputedValue, ToCss)]
+pub enum ListStyleType {
+    /// <counter-style> | none
+    CounterStyle(CounterStyleOrNone),
+    /// <string>
+    String(String),
```

Comment:
If you also want to save a bit of memory (assuming `CounterStyleOrNone` is smaller than `String`, which I haven't checked), you could use `Box<str>`.

Reply:
What's the best way to check this sort of thing? Here's the hacky approach I took...

I edited `tests/unit/style/size_of.rs` to add these lines:

``` rust
use style::values::specified::list::ListStyleType;
size_of_test!(test_size_of_list_style_type, ListStyleType, 0);
```

Then edited the source to get rid of various `#[cfg(feature = "gecko")]` lines. Then ran the test with `String(String)` and `String(Box<str>)`.

In both cases the size was 40 bytes, so I guess `CounterStyleOrNone` is larger than `String`.

Reply:
Yeah, string should be 32 bytes. I usually just do the math manually. Since `CounterStyleOrNone` contains a `Vec<..>` (32 bytes) and the enum tag, that makes sense.